vm/x86/x86emu/src/emulator.rs

```diff
@@ -462,13 +462,13 @@ impl<'a, T: Cpu> Emulator<'a, T> {
     }
 
     /// Writes memory to the given segment:offset.
-    async fn write_memory(
+    pub async fn write_memory(
         &mut self,
         segment: Register,
```

Comment:
Why don't these methods just take a `Segment` instead of a `Register` that they assert is a segment and then convert into a `Segment`?